Fix/pacmod tracking

sumanth-nirmal opened the pull request:
## Status
**PRODUCTION / DEVELOPMENT**

## Description
This fixes the velocity computation in tracking

## Steps to Test or Reproduce
Ego car should not break suddenly when a car cut in.

sumanth-apexai committed:
fix the velocity computation

sumanth-apexai committed:
added current velocity in computation

dejanpan commented on ros/src/computing/planning/motion/packages/astar_planner/nodes/velocity_set/libvelocity_set.h:
```diff
@@ -233,13 +240,59 @@ class ObstacleTracker
-    double v = kf_.update(dx/dt.toSec());
-    v = (v > moving_thres_) ? v : 0.0;
+    double v;
+    ros::Duration dt = time_buf_[1]-time_buf_[0]; double dx;
```
double dx; => into new line

sumanth-nirmal replied:
This is just for debug, I will change it when removing the debug print's

dejanpan commented on ros/src/computing/planning/motion/packages/astar_planner/nodes/velocity_set/libvelocity_set.h:
```diff
@@ -233,13 +240,59 @@ class ObstacleTracker
+            }
+            else
+            {
+                v = prev_velocity_;
```
in this case we just keep going with whatever the v was before we started getting negative dx?

sumanth-nirmal replied:
correct, we keep the prev_velocity_

dejanpan commented on ros/src/computing/planning/motion/packages/astar_planner/nodes/velocity_set/velocity_set.cpp:
```diff
@@ -649,6 +652,7 @@ int main(int argc, char** argv)
 
     int obstacle_waypoint = -1;
     double obstacle_velocity = 0.0;
+    current_velocity_ = vs_path.getCurrentVelocity();
```
are we back to initialization from waypoints' velocity?

sumanth-nirmal replied:
this is ego car current velocity used to initialize the `prev_velocity_` variable.

aohsato reviewed (approved):
This modification works well under some rosbags. Great job.

sumanth-apexai committed:
increased the tracking_moving_thres to 10kmph

aohsato committed:
Fix tracking to initialize when big position jump occur

sumanth-nirmal committed:
added rule to handle negative dx when front of the car is detected first and then back

sumanth-apexai committed:
Merge branch 'feature/pacmod_apex' into fix/pacmod_tracking

sumanth-nirmal committed:
using mean of points to get obstacle

sumanth-nirmal committed:
truncate negative raw velocity to zero
disable tracking if its with in hard stop distance
added soft stop distance based on velocity

sumanth-nirmal committed:
hold the stopwaypoint when we are in hard stop distance
update KF but ignore the estimate when negative dx is allowed

sumanth-nirmal committed:
removed the debug output
resolved the warnings

dejanpan commented on ros/src/computing/planning/motion/packages/astar_planner/nodes/velocity_set/libvelocity_set.h:
```diff
@@ -266,23 +311,50 @@ class ObstacleTracker
     position_buf_.push_back(position_);
     time_buf_.push_back(time_);
 
+    // this is to handle if a car cuts in front of us of we are already tracking a car way ahead of us
```
of we are => and if we

dejanpan committed:
typo
